### DISCUSSION TITLE
Overflow="fold" makes console path unusable for navigation

### DISCUSSION THREAD

**Discussion by pnsak** [Textualize/rich#1742] [Q&A]
Using overflow="fold" makes printed path unusable for navigation through Ctrl+RightClick.
![image](https://user-images.githubusercontent.com/8767646/144894000-d4dd2630-bed4-494e-a4ca-8cf1b659fc01.png)

```
from rich.console import Console
from rich.table import Table
from rich.markup import escape

table = Table(title="Star Wars Movies")

table.add_column("Data", justify="right", style="cyan", no_wrap=True)
table.add_column("RunPath", style="magenta", overflow="fold", width=37)
table.add_column("Status", justify="right", style="green")

table.add_row(
    "May 25, 2018", str(Path.cwd()), "DONE",
)

console = Console()
console.print(table)
```

If I increase the column width to accommodate the path, then Ctrl+RightClick is working. Is there anyway I could get this functionality with wrap?

**Comment by willmcgugan** [comment]
You can style a string as a link with a file link (`"[link=file:///c/foo/bar/]"`) but not all terminals support links.

**Reply by pnsak** [reply]
Hi Will,
Thanks for your response. The basic difference I observe is console.print vs native print in python.
When I print a long path using "print" it's clickable even if some of the path goes into the next line, however, when I use "console.print" the path gets kind of separated into two independent lines which did not allow navigation.
Additionally, native print's printed path wraps automatically when I resize terminal size but not with console.print.

Thank you for making this great library

**Reply by willmcgugan** [reply]
It's your terminal software that makes a path clickable. If your path is within a table and there is not enough space, then Rich has no option but to truncate or wrap it -- which will break the terminals parsing of the path. That's not something Rich can control, unless your terminal supports explicit links.

If you want to do a plain print and you don't want a path to wrap, you can set `soft_wrap=True`. See the docs for more information about soft wrapping.
